lib/screens/terms_and_conditions/screen.dart

```diff
@@ -121,11 +121,11 @@ class _TermsAndConditionsScreenState extends State<TermsAndConditionsScreen> {
 
   Function()? _onAcceptButtonPressed(BuildContext context) {
     if (isAccepted) {
-      return () async {
+      return () {
         final tac = AcceptedTermsAndConditions(
           version: widget.validTermsAndConditions.version,
         );
-        await context.read<TermsAndConditionAcceptance>().userAccepted(tac);
+        context.read<TermsAndConditionAcceptance>().userAccepted(tac);
```

Comment:
This reverts https://github.com/Concordium/concordium-wallet/pull/12#discussion_r1377305866 because the persist call is being handled in an `onChange` observer method which is called after setting the state.

But I think this is actually the right approach as we don't want async actions to block the UI. Rather, the async completion should push its status somewhere which might then trigger a UI update.